udpSource/README.md

```diff
@@ -0,0 +1,191 @@
+	* pattern: Required. The regex pattern that will be used to parse the incoming data. The parser will use the first match that appears in the data. See java.util.regex.Pattern for specifics on what constitutes a valid pattern.
+	* locations: Required. An array of the locations in which to place the capture groups from pattern.
+
+
+## <a name="mapTransformer" id="mapTransformer"></a>MapTransformer
+
+MapTransformer is designed to deal with getting and putting for nested Maps. 
+
+### Locations
+A location is written as period separated key names starting from the inital map, e.g. "level1.level2.level3" would be the location of "Hello World" in the Map represented by the following JSON object.
+```
+{
+    level1: {
+        level2: {
+            level3: {
+                "Hello World"
+            }
+        }
+    }
+}
+```
+
+### Transformations
+A transformation copies an object from a location in the input Map to a location in the output Map, overwriting anything in the path for the output map. If the input location is invalid or null, nothing is written in the output Map. A transformation is written as an array with the location in the input as the first element and the desired location in the output as the second element.
+Example: Using the transformation ["level1.level2", "lvl1.lvl2"] on the input map
+```
+{
+	level1: {
+		level2: "Hello World"
+	}
+}
+```
+and the output map
+```
+{
+	lvl1: "Sir Not-Appearing-In-This-Result",
+	string: "Hello"
+}
+```
+would result in the output map looking like
+```
+{
+	lvl1: {
+		lvl2: "Hello World"
+	},
+	string: "Hello"
+}
+```
+since the value in lvl1 was overwritten by the Map created to contain lvl2 and the transformed value.
+<br>
+Additionally, if the output map had instead been
+```
+{
+	lvl1: {
+		lvl2: {
+			lvl3: "Sir Not-Appearing-In-This-Result"
+		}
+	},
+	string: "Hello"
+}
+```
+then the resulting map would be exactly the same, as the Map in lvl2 would have been overwritten by the new value.
+
+## Licensing
+The source code in this library is licensed under the the [MIT License](https://opensource.org/licenses/MIT).
+
+This library uses several licensed libraries, some of which are more restrictive than the MIT license. 
+okhttp3 and jackson-databind are both licensed under [Apache Version 2.0 License](http://www.apache.org/licenses/LICENSE-2.0). 
+slf4j is licensed under [terms](https://www.slf4j.org/license.html) identical to the [MIT License](https://opensource.org/licenses/MIT).
+logback is licensed under either the [Eclipse Public License v1.0](http://www.eclipse.org/legal/epl-v10.html) or [Lesser GPL v2.1](www.gnu.org/licenses/old-licenses/lgpl-2.1.html), at the licensee's discretion. This library is optional, and can be replaced with another slf4j implementation if you wish to reduce the license restrictions. To do so, replace the line `compile "ch.qos.logback:logback-classic:1.2.3"` with your own import, or remove that line for no logging.
```

Comment:
1) ...replace the line... (need to note that you replace this in the `build.gradle` file.

Reply:
@The review comment, I moved the dependency from extjsdk to here. If I'm remembering correctly the reason that it was in exjsdk was that I wanted to see the statements in the tests so I could debug it, and then I forgot to remove it.